bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/ThermostatHandler.java

```diff
@@ -0,0 +1,241 @@
+                    updateProperties(properties);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand called for channel: {}, command: {}", channelUID.getAsString(), command);
+        final OmnilinkBridgeHandler bridgeHandler = getOmnilinkBridgeHandler();
+        Optional<TemperatureFormat> temperatureFormat = Optional.empty();
+
+        if (!(command instanceof DecimalType)) {
+            logger.warn("Invalid command: {}, must be DecimalType", command);
+            return;
+        }
+        if (bridgeHandler != null) {
+            temperatureFormat = bridgeHandler.getTemperatureFormat();
+            if (!temperatureFormat.isPresent()) {
+                logger.warn("Receieved null temperature format!");
+                return;
+            }
+        } else {
+            logger.warn("Could not connect to Bridge, failed to get temperature format!");
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_THERMO_SYSTEM_MODE:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_SYSTEM_MODE.getNumber(),
+                        ((DecimalType) command).intValue(), thingID);
+                break;
+            case CHANNEL_THERMO_FAN_MODE:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_FAN_MODE.getNumber(), ((DecimalType) command).intValue(),
+                        thingID);
+                break;
+            case CHANNEL_THERMO_HOLD_STATUS:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_HOLD_MODE.getNumber(),
+                        ((DecimalType) command).intValue(), thingID);
+                break;
+            case CHANNEL_THERMO_HEAT_SETPOINT:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_HEAT_LOW_POINT.getNumber(),
+                        temperatureFormat.get().formatToOmni(((DecimalType) command).intValue()), thingID);
+                break;
+            case CHANNEL_THERMO_COOL_SETPOINT:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_COOL_HIGH_POINT.getNumber(),
+                        temperatureFormat.get().formatToOmni(((DecimalType) command).intValue()), thingID);
+                break;
+            case CHANNEL_THERMO_HUMIDIFY_SETPOINT:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_HUMDIFY_POINT.getNumber(),
+                        TemperatureFormat.FAHRENHEIT.formatToOmni(((DecimalType) command).intValue()), thingID);
+                break;
+            case CHANNEL_THERMO_DEHUMIDIFY_SETPOINT:
+                sendOmnilinkCommand(OmniLinkCmd.CMD_THERMO_SET_DEHUMIDIFY_POINT.getNumber(),
+                        TemperatureFormat.FAHRENHEIT.formatToOmni(((DecimalType) command).intValue()), thingID);
+                break;
+            default:
+                logger.warn("Unknown channel for Thermostat thing: {}", channelUID.getAsString());
```

Comment:
```suggestion
                logger.warn("Unknown channel for Thermostat thing: {}", channelUID);
```

`ChannelUID` properly implements `toString`